### DISCUSSION TITLE
Unable to add transformRequest to an endpoint

### DISCUSSION THREAD

**Discussion by existe-deja** [nuxt-community/auth-module#1616] [Q&A]
### Version
module: 5.0.0-1618898588.9655d0e
nuxt: 2.15.4

#### [mode](https://nuxtjs.org/api/configuration-mode): 
 - [x] universal
 - [ ] spa

### Nuxt configuration
```
login: {
  url: '/oauth/token/',
  method: 'post',
  headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
  transformRequest: [(data, headers) => {
    console.log('transformRequest', data)
    return data
  }]
}
```

## Reproduction
https://codesandbox.io/s/pedantic-waterfall-1ze8d?file=/nuxt.config.js

#### What is expected?
I should be able to transform the request before it's sent.
According to the doc:
> Each endpoint is used to make requests using axios. They are basically extending Axios Request Config.
But I can't add a request transformer specified in https://github.com/axios/axios#request-config

I'm trying to tranform the json data into a www-form-urlencoded data (stringified).
Poke https://github.com/nuxt-community/auth-module/issues/977

#### What is actually happening?
An error pops before the request is made. Specifying `fn is not a function`.

#### Steps to reproduce
Add a transformRequest Array in the endpoint definition. Or go to the reproduction link.

## Checklist
* [x]  I have tested with the latest Nuxt version and the issue still occurs
* [x]  I have tested with the latest module version and the issue still occurs
* [x]  I have searched the issue tracker and this issue hasn't been reported yet

**Comment by bmulholland** [comment]
Axios was recently updated, does that resolve this?

**Comment by GMartigny** [comment]
I have what seams to be the same issue with the latest version possible.

auth-next: 5.0.0-1648802546.c9880dc
nuxt/axios: 5.13.6
nuxt: 2.15.0

[Repro repo](https://codesandbox.io/s/magical-shockley-56bosl)

You can see that in the page, the `transformRequest` option is `[null]` for some reason.

**Comment by GMartigny** [comment]
I just saw that in the file `templates/plugin.js`, the options passed down to the scheme are [`JSON.stringify`-ed](https://github.com/nuxt-community/auth-module/blob/dev/templates/plugin.js#L20).
This means that only primitive type are allowed (so not functions). This prevent users from defining functions as value for any options.

**Comment by tai-tran-tan** [comment]
I got the same problem when trying to convert body data for 'application/x-www-form-urlencoded'
`transformRequest: [(data, headers) => new URLSearchParams(data)]`
TypeError: Cannot read properties of null (reading 'call')
    at transform (transformData.js?c401:18:1)
    at Object.forEach (utils.js?c532:245:1)
    at Object.transformData (transformData.js?c401:17:1)
    at dispatchRequest (dispatchRequest.js?5270:35:1)
    at async RefreshScheme.login (runtime.mjs?edb1:1193:1)

Versions:
"@nuxtjs/auth-next": "5.0.0-1648802546.c9880dc",
    "@nuxtjs/axios": "^5.13.6",
    "nuxt": "^2.12.2",

**Comment by mhshakouri** [comment]
would be happy if anyone got any idea for a workaround on this

**Reply by tai-tran-tan** [reply]
Mine is to convert all neccessary data into URLSearchParams
``` 
const data = new URLSearchParams({
        ...account,
        ...this.$config.keycloak,
      });
this.$auth.loginWith("local", { data })`
